packages/server/lib/browsers/firefox.ts

```diff
@@ -349,9 +349,14 @@ toolbar {
 let browserCriClient
 
 export function _createDetachedInstance (browserInstance: BrowserInstance, browserCriClient?: BrowserCriClient): BrowserInstance {
+  if (!browserInstance.pid) {
+    throw new Error('Browser PID is undefined. Browser process failed to spawn.')
+  }
+
+  const browserInstancePid = browserInstance.pid
```

Comment:
With 16.x, PID has been changed to `number | undefined`. It seems that it was a bug. [It was `number | undefined` at 14.x doc](https://nodejs.org/docs/latest-v14.x/api/child_process.html). 

I changed the type of `BrowserInstance` and throw Error when something went wrong.